### PR TITLE
OAK-9054-Fix of OAK-8930 leading to increase in time to get blobs from Azure.

### PR DESCRIPTION
**Pull request by bhardwajrahul20**
1. Adding break statement to break the loop if call to listBlobsSegmented is successful.
2. Doing refactoring to reduce cyclomatic complexity.

### CODE REVIEW

**Review by chibulcuteanu — APPROVED** [review]
LGTM

File: oak-segment-azure/src/main/java/org/apache/jackrabbit/oak/segment/azure/AzureUtilities.java [hunk 1]
```diff
@@ -159,6 +134,38 @@ public static CloudBlobDirectory cloudBlobDirectoryFrom(String connection, Strin
...
+                        null);
+                break;  //we have the results, no need to retry
+            } catch (StorageException | URISyntaxException e) {
+                lastException = new IOException(e);
+                try {
+                    Thread.sleep(sleep); //Sleep and retry
+                } catch (InterruptedException ex) {
+                    log.warn("Interrupted", e);
+                }
+                continue;
```
Comment (thomasmueller) [inline oak-segment-azure/src/main/java/org/apache/jackrabbit/oak/segment/azure/AzureUtilities.java]: I don't think "continue" is needed her

**Review by thomasmueller — APPROVED** [review]

File: oak-segment-azure/src/main/java/org/apache/jackrabbit/oak/segment/azure/AzureUtilities.java (hunk 1 above)
Comment (bhardwajrahul20) [reply oak-segment-azure/src/main/java/org/apache/jackrabbit/oak/segment/azure/AzureUtilities.java]: +1. I think, I missed it in refactoring.

**Comment by mreutegg** [comment]
This was committed to SVN.

**Pull request closed without merging**